imports.go

```
@@ -1,135 +1,167 @@
 package main
 
 import (
-	"go/build"
-	"log"
+	"bufio"
+	"encoding/json"
+	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
 	"sort"
 	"strings"
 )
 
-func recursiveParseImports(
-	imports map[string]bool, path string, cwd string,
-) error {
-	if path == "C" {
-		return nil
+func parseImports(recursive bool, testDependencies bool) ([]string, error) {
+	var imports []string
+	packages, err := listPackages()
+	if err != nil {
+		return imports, fmt.Errorf("error listing packages: %s", err)
 	}
 
-	// catch internal vendoring in net/http since go 1.7
-	if strings.HasPrefix(path, "golang_org/") {
-		return nil
-	}
+	ensureDependenciesExist(packages, testDependencies)
 
-	pkg, err := build.Import(path, cwd, build.IgnoreVendor)
+	imports, err = calculateDependencies(packages, recursive, testDependencies)
 	if err != nil {
-		return err
+		return imports, err
 	}
 
-	if path != "." {
-		standard := false
+	imports = filterPackages(imports)
+
+	sort.Strings(imports)
+
+	return imports, nil
+
+}
+
+func calculateDependencies(packages []string, recursive, testDependencies bool) ([]string, error) {
+	var deps, imports, testImports, testDeps []string
+
+	for _, pkg := range packages {
+		data, err := list(pkg)
+		if err != nil {
+			return imports, nil
```

why error is ignored?